### DISCUSSION TITLE
Django, Error: HTTP 200 (OK), should be 206 for range requests

### DISCUSSION THREAD

**Discussion by BioOmics** [GMOD/jbrowse-components#4021] [Q&A] [answered]
Recently, i want to run `JBrowse 2` on the Django server,

Everything is fine, such as i can open the `index.html` and load the basic files (such as `genome.fa.fai`), and open a basic circos plot....
However, i can't load the files (such as `gff3` or `genome.fa`) from Django server.

The **error** looks like: 
`Error: HTTP 200 (OK) when fetching http://localhost:8080/static/jbrowseapp/Data/genome.fa bytes 72482816-72744959 (should be 206 for range requests)`

i have seen your FAQ here: [what-web-server-do-i-need-to-run-jbrowse-2](https://jbrowse.org/jb2/docs/faq/#what-web-server-do-i-need-to-run-jbrowse-2)

I know use `npx serve -C` can start a web server at `http://localhost:3000`. However, it can't be uesd in Django
**So, now, i really don't konw how to solve this problem.**
**How can i get small slices of large binary data files by Django only, as i don't use any Apache, Nginx or npx....**

---

If you konw Django, i can share you my configs below:

`views.py`
```python
from django.shortcuts import render

def jbrowse(request):
    return render(request,"jbrowseapp/jbrowse.html")
```

`urls.py`
```python
from django.urls import path
from jbrowseapp import views

urlpatterns = [
    path('jbrowse/',views.jbrowse, name='jbrowse'),
]
```

`jbrowse.html`
```html
{% load static %}
<!doctype html>
<html lang="en">
<head>
    <meta charset="utf-8"/>
    <link rel="shortcut icon" href="{% static 'jbrowseapp/favicon.ico' %}"/>
    <meta name="viewport" content="width=device-width,initial-scale=1"/>
    <meta name="theme-color" content="#000000"/>
    <meta name="description" content="A fast and flexible genome browser"/>
    <link rel="manifest" href="{% static 'jbrowseapp/manifest.json' %}"/>
    <title>JBrowse</title>
    <script defer="defer" src="{% static 'jbrowseapp/static/js/main.f49dd59b.js' %}"></script>
</head>
<body style="overscroll-behavior:none">
    <noscript>You need to enable JavaScript to run this app.</noscript>
    <div id="root"></div>
</body>
</html>
```
**I really really need your help! THANK YOU!!**

**Comment by cmdcolin** [comment, marked as answer]
@BioOmics hi there. When I last evaluated this issue, it was actually not possible to get Django to serve Range requests. This is something that jbrowse relies on for accessing large data files. I would recommend hosting the data files on a server separately from Django https://github.com/cmdcolin/django-jbrowse2-nonworking-example
